src/OSPSuite.Infrastructure.Import/Services/Importer.cs

```diff
@@ -173,16 +178,25 @@ public MappingProblem CheckWhetherAllDataColumnsAreMapped(ColumnInfoCache dataCo
 
          return new MappingProblem()
          {
+            //all the mandatory mappings that have not been mapped to a column
             MissingMapping = dataColumns
                .Where(col => col.IsMandatory && subset.All(cm =>
                   cm.MappedColumn.Name != col.Name)).Select(col => col.Name)
                .ToList(),
-            MissingUnit = subset.Where(cm => cm.MappedColumn.Unit.SelectedUnit == UnitDescription.InvalidUnit && (cm.MappedColumn.ErrorStdDev == null || cm.MappedColumn.ErrorStdDev.Equals(Constants.STD_DEV_ARITHMETIC))).Select(cm => cm.MappedColumn.Name)
+            //all the mappings where the unit is undefined and 1. the error is not geometric or 2. the mapped column is empty
+            MissingUnit = subset
+               .Where(
+                  cm => cm.MappedColumn.Unit.SelectedUnit == UnitDescription.InvalidUnit && 
```

Comment:
This is so hard to read. This needs to be changed.
It looks to me like this could be a property of the MappedColumn instead of constantly checking properties on the object

Reply:
@msevestre moved it to the Column class